Wifi now component

motwok opened the pull request:
## Description:
wifi_now esphome component - communicate directly to other ESP device via ESP-Now

**Related issue (if applicable):** fixes <link to issue>
[ESP-Now integration #275](https://github.com/esphome/feature-requests/issues/275)

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1290

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

motwok committed:
wifi now compnent documentation

motwok committed:
add wifi_now to index.rst

destos committed:
editing language on wifi_now component docs

destos committed:
linting wifi_now docs

motwok committed:
fix svg name

motwok committed:
fix png filename

motwok committed:
try to fix seo

motwok committed:
fix lint messages

motwok committed:
try to fix lint

motwok committed:
fixing lint meassage

motwok committed:
fixing lint message

motwok committed:
fixing lint messages

motwok committed:
fixing lint message

motwok committed:
fixing lint message

jesserockz commented on index.rst:
```diff
@@ -354,6 +354,7 @@ Misc Components
     Exposure Notifications, components/exposure_notifications, exposure_notifications.png
     RTTTL Buzzer, components/rtttl, buzzer.jpg
     Prometheus, components/prometheus, prometheus.svg
+    Wifi Now, component/wifi_now, wifi_now.svg
```
```suggestion
    Wifi Now, components/wifi_now, wifi_now.svg
```

motwok replied:
thanks

motwok committed:
fix lint messages

motwok committed:
fixing lint messages

motwok committed:
Update index.rst

Co-authored-by: Jesse Hills <3060199+jesserockz@users.noreply.github.com>

motwok committed:
fix internal refs

motwok committed:
fixing injection sample

motwok committed:
fixing typp

motwok committed:
Update wifi_now.rst

motwok committed:
Update wifi_now.rst

motwok committed:
Update wifi_now.rst

motwok committed:
Update wifi_now.rst

motwok committed:
Merge branch 'next' into wifi_now_component

motwok committed:
Merge branch 'next' into wifi_now_component

motwok commented:
Push for first annual

daztucker committed:
Add suggestion for reducing DTIM to improve WiFi

This change improved my use case from ~50% packet loss to roughly zero and allows OTA updates to work reliably.

RebbePod committed:
Add See Also Base Text Sensor

Perhaps also remove the reference to config right above

RebbePod committed:
Add See Also Base Text Sensor

ludrao committed:
Added a DIY sample configuration

The linked repo also contain a quick description on how to transform a noname Tuya switch into a device that actually run ESPHome

ludrao committed:
Added a second config sample

aquaticus committed:
Update rotary encoder doc

Added extra options example for encoder pin configuration.

gonzalop committed:
Adds missing update_interval config variable

gonzalop committed:
Update st7735.rst

spacemanspiff2007 committed:
Clearified heartbeat

abmantis committed:
Add docs to pzemac on how to change its address

Most tutorials or suggestions online point to using either the official PZEM tool (windows only) or Tasmota. 
With the suggested snipped it is possible to do it using ESPHome.

abmantis committed:
Use a better id

abmantis committed:
Fix underline lint

stegm committed:
Fix options and use smaller/better example.

stegm committed:
Typos.

oxan committed:
Update components/display/st7735.rst

DAVe3283 committed:
Fix MAX31856 image

The wrong image was selected, likely due to a simple typo.

jesserockz committed:
Bump version to 2022.1.0b1

jesserockz committed:
Update changelog for 2022.1.0b1

jesserockz committed:
Update supporters for 2022.1.0b1

jesserockz committed:
Merge pull request #1794 from esphome/bump-2022.1.0b1

2022.1.0b1

jesserockz committed:
Remove the changelog toctrees

flaviut committed:
Merge pull request #1795 from esphome/remove-changelog-toctrees

Remove the changelog toctrees

jesserockz committed:
Merge branch 'current' into beta

jesserockz committed:
Remove toctree from 2022.1 changelog

flaviut committed:
Merge pull request #1777 from stegm/bugfix_modbus_textsensor

Fix options and use smaller/better example.

flaviut committed:
Merge pull request #1772 from abmantis/patch-1

Add docs to pzemac on how to change its address

flaviut committed:
Merge pull request #1770 from spacemanspiff2007/patch-1

Clearified heartbeat

flaviut committed:
Merge pull request #1751 from aquaticus/patch-1

Update rotary encoder doc

flaviut committed:
Merge pull request #1720 from RebbePod/patch-5

Add See Also Base Text Sensor

flaviut committed:
Merge pull request #1721 from RebbePod/patch-6

Add See Also Base Text Sensor

flaviut committed:
Merge pull request #1580 from daztucker/patch-1

Add suggestion for reducing DTIM to improve WiFi

flaviut committed:
Merge pull request #1747 from ludrao/patch-1

Added a DIY sample configuration

flaviut committed:
Merge pull request #1757 from gonzalop/patch-1

Adds missing update_interval config variable

flaviut committed:
Merge pull request #1792 from DAVe3283/max31856-image

Fix MAX31856 image

jesserockz committed:
Merge branch 'current' into beta

jesserockz committed:
Bump version to 2022.1.0b2

jesserockz committed:
Update changelog for 2022.1.0b2

jesserockz committed:
Update supporters for 2022.1.0b2

jesserockz committed:
Merge pull request #1797 from esphome/bump-2022.1.0b2

2022.1.0b2

jesserockz committed:
Add select and button to REST API docs (#1805)

jesserockz committed:
Add number setting to web_server (#1804)

jesserockz committed:
Merge branch 'current' into beta

jesserockz committed:
Bump version to 2022.1.0b3

jesserockz committed:
Update changelog for 2022.1.0b3

jesserockz committed:
Update supporters for 2022.1.0b3

jesserockz committed:
Merge pull request #1807 from esphome/bump-2022.1.0b3

2022.1.0b3

jesserockz committed:
Add discord and forums to keeping up (#1808)

jesserockz committed:
Merge branch 'current' into beta

jesserockz committed:
Bump version to 2022.1.0b4

jesserockz committed:
Update changelog for 2022.1.0b4

jesserockz committed:
Update supporters for 2022.1.0b4

balloob committed:
Add section about ESPHome Web

jesserockz committed:
Merge branch 'beta' into bump-2022.1.0b4

jesserockz committed:
Merge branch 'current' into beta

jesserockz committed:
Merge pull request #1811 from esphome/bump-2022.1.0b4

2022.1.0b4

jesserockz committed:
A few updates on canbus page (#1813)

jesserockz committed:
Bump version to 2022.1.0

jesserockz committed:
Update supporters for 2022.1.0

jesserockz committed:
Merge pull request #1814 from esphome/bump-2022.1.0

2022.1.0

jesserockz committed:
Fix pronoun (#1815)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

sbur83 committed:
Update tuya.rst (#1812)

mmakaay committed:
Removed copy/paste error from ina260 doc

mmakaay committed:
Merge pull request #1817 from esphome/not-afraid-to-edit-on-github

Removed copy/paste error from ina260 doc

hubikj committed:
Update menu path for compiled firmware download (#1816)

jesserockz committed:
Fix gpio page mentioning PWM (#1818)

jesserockz committed:
Bump version to 2022.1.1

jesserockz committed:
Update changelog for 2022.1.1

jesserockz committed:
Update supporters for 2022.1.1

jesserockz committed:
Fix

jesserockz committed:
Merge pull request #1819 from esphome/bump-2022.1.1

2022.1.1

bgulla committed:
fix typo in it.invert_on_off() (#1821)

jesserockz committed:
Update dashboard screenshots (#1791)

Co-authored-by: Flaviu Tamas <me@flaviutamas.com>

stonie08 committed:
Update qmc5883l.rst (#1828)

fixed minor typo in the documentation

JeeCee1 committed:
Update xiaomi_ble.rst (#1825)

Added ´´´The easiest method to retrieve the bindkey from the cloud is to use the Cloud Tokens Extractor, written by one of Home Assistant users. If you prefer to not use the executable, read here.´´ and a link under "See Also"

JeeCee1 committed:
Update esp32_ble_tracker.rst (#1824)

Added ``Defaults to 30ms`` to ``window`` parameter.

eduperez committed:
Fix temperature-related variable names (#1820)

edgimar committed:
add sample config for Tuya Smart Life RGBW controller (#898)

swifty99 committed:
Clarification of the behavior of calibraty_linear filter (#1701)

spacemanspiff2007 committed:
Specified how the sensor values get populated (#1809)

* Specified how the sensor values get populated

* Update template.rst

* Update template.rst

Reword things a little

Co-authored-by: Flaviu Tamas <me@flaviutamas.com>

leakypixel committed:
Update mqtt to reflect changes in 2022.1 (#1826)

* Update mqtt to reflect changes in 2022.1

`root` is no longer passed by reference; update the documentation to reflect this.

* Fix long-broken subscribe_json call

Co-authored-by: Flaviu Tamas <tamasflaviu@gmail.com>

flaviut committed:
Remove recommendation from calibrate_linear (#1830)

Requested here: https://github.com/esphome/esphome-docs/pull/1701#discussion_r789486034

flaviut committed:
Improve documentation around installing ESPHome and connecting devices to computers (#1787)

* Begin for-dummies programming guide

* Delete getting-started-cli content

* Seperate connection guide

* Refactor instructions

Have the Getting started guides link to the correct subguides when they
need to explain how to do a common thing.

* Apply suggestions from code review

Co-authored-by: Oxan van Leeuwen <oxan@oxanvanleeuwen.nl>

* Add physical connection note to hassio guide

* Improve page titles

* Mention lack of tested Mac instructions

* Fix typos

Co-authored-by: Oxan van Leeuwen <oxan@oxanvanleeuwen.nl>

oxan committed:
Add note that buttons are unidirectional (#1831)

flaviut committed:
Add virtualenvs to sphinx ignore (#1840)

We don't want to try and include these directories in our output

flaviut committed:
Process files with netlify (#1841)

I'm not sure if this will actually work, but if it does, it'll reduce
the size of a bunch of the image files automatically & not require us to
worry about that anymore.

It'll also combine all our JS & CSS files into a single bundled request.

flaviut committed:
Revert "Process files with netlify (#1841)" (#1842)

This reverts commit 3d78aecc4732a0c031be2d2d8ceec3d192e3f119.

Turns out that there's a time limit on builds, and sphinx output has a
whole lot of JS that gets bundled and minified.

We end up running up against the time limit and the whole build fails.
Therefore, revert it.

dudanov committed:
Use Buttons instead of switches in midea example (#1845)

jesserockz committed:
Bump version to 2022.1.2

jesserockz committed:
Update changelog for 2022.1.2

jesserockz committed:
Update supporters for 2022.1.2

jesserockz committed:
Merge pull request #1847 from esphome/bump-2022.1.2

2022.1.2

flaviut committed:
Add Github UI instructions for docs (#1838)

flaviut committed:
Add live reloading (#1786)

Replace the existing webserver target with a live-reloading target.

Adds a dependency on https://github.com/executablebooks/sphinx-autobuild

nagyrobi committed:
Clarification for MCP23017 (#1854)

trickv committed:
Update link to PlatformIO boards list (#1852)

martgras committed:
remove incomple mqtt component (#1848)

ssieb committed:
Fix misleading status led note (#1858)

OttoWinter committed:
Optimize images displayed on homepage (#1860)

jesserockz committed:
Bump version to 2022.1.3

jesserockz committed:
Update changelog for 2022.1.3

jesserockz committed:
Update supporters for 2022.1.3

jesserockz committed:
Merge pull request #1874 from esphome/bump-2022.1.3

2022.1.3

AshtonKem committed:
Update ble_client.rst (#1869)

jesserockz committed:
Revert "Update ble_client.rst (#1869)" (#1884)

This reverts commit 43d5bf96f4242f62e6c26be3e20a88aee227db74.

jesserockz committed:
Update home assistant addon repo (#1890)

jesserockz committed:
Bump version to 2022.1.4

jesserockz committed:
Update changelog for 2022.1.4

jesserockz committed:
Update supporters for 2022.1.4

jesserockz committed:
Merge pull request #1891 from esphome/bump-2022.1.4

2022.1.4

bennop committed:
Added missing verb (#1894)

ptoal committed:
Update Home Assistant getting started guide (#1889)

depuits committed:
fixed on_boot wrongly positioned in modbus controller example (#1882)

patrickcollins12 committed:
Update diy.rst (#1881)

depuits committed:
Switch modbus controller register_type (#1883)

Co-authored-by: Jesse Hills <3060199+jesserockz@users.noreply.github.com>

jesserockz committed:
Some docs tidy up (#1895)

motwok committed:
Merge branch 'esphome:current' into wifi_now_component

nagyrobi commented:
Closing as the corresponding PR is outdated by the author: https://github.com/esphome/esphome/pull/1290#issuecomment-1049138410
